tests/prague/eip7676_address_space_extension/test_ase_opcodes.py

```diff
@@ -0,0 +1,219 @@
+"""
+Execution of CALLF, RETF opcodes within EOF V1 containers tests
+"""
+
+import pytest
+
+from ethereum_test_tools import (
+    Account,
+    Address,
+    Environment,
+    StateTestFiller,
+    TestAddress,
+    Transaction,
+)
+from ethereum_test_tools.common.conversions import BytesConvertible, to_bytes
+from ethereum_test_tools.eof.v1 import Container, Section
+from ethereum_test_tools.eof.v1.constants import NON_RETURNING_SECTION
+from ethereum_test_tools.vm.opcode import Opcodes as Op
+
+from .spec import EOF_FORK_NAME
+
+REFERENCE_SPEC_GIT_PATH = "EIPS/eip-7676.md"
+REFERENCE_SPEC_VERSION = (
+    "00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+)
+
+pytestmark = pytest.mark.valid_from(EOF_FORK_NAME)
+
+
+@pytest.mark.parametrize(
+    "target_address",
+    (
+        b"",
+        b"\x10\x00",
+        b"\x78" * 20,
+        b"\xff" * 20,
+        b"\x01" + (b"\x00" * 20),
+        b"\x5a" * 28,
+        b"\x5a" * 32,
+        b"\xff" * 32,
+    ),
+    ids=["zero", "short", "mid20", "max20", "minAse", "midAse", "fullAse", "maxAse"],
+)
+@pytest.mark.parametrize("target_account_type", ("empty", "EOA", "Contract"), ids=lambda x: x)
+def test_address_space_extension(
+    state_test: StateTestFiller,
+    target_address: BytesConvertible,
+    target_account_type: str,
+):
+    """
+    Test contacts with possibly extended address and fail if address is too large
+    """
+    env = Environment()
+
+    address_bytes = to_bytes(target_address)
+    ase_address = len(address_bytes) > 20
+    if ase_address and address_bytes[0] == b"00":
+        raise ValueError("Test instrumentation requires target addresses trim leading zeros")
+
+    pre = {
+        TestAddress: Account(
+            balance=1000000000000000000000,
+            nonce=1,
+        ),
+        Address(0x100): Account(
+            code=(
+                Op.MSTORE(0, Op.PUSH32(address_bytes))
+                + Op.SSTORE(0, Op.CALL(50000, 0x200, 0, 0, 32, 0, 0))
+                + Op.SSTORE(1, Op.CALL(50000, 0x300, 0, 0, 32, 0, 0))
+                + Op.SSTORE(2, Op.CALL(50000, 0x400, 0, 0, 32, 0, 0))
+                + Op.SSTORE(3, Op.CALL(50000, 0x500, 0, 0, 32, 0, 0))
+                + Op.SSTORE(4, Op.CALL(50000, 0x201, 0, 0, 32, 0, 0))
+                + Op.SSTORE(5, Op.CALL(50000, 0x301, 0, 0, 32, 0, 0))
+                + Op.SSTORE(6, Op.CALL(50000, 0x401, 0, 0, 32, 0, 0))
+                + Op.SSTORE(7, Op.CALL(50000, 0x501, 0, 0, 32, 0, 0))
+                + Op.SSTORE(8, Op.CALL(50000, 0x601, 0, 0, 32, 0, 0))
+                + Op.STOP()
```

Comment:
storage slots 1...8 can also be variables with the names describing the meaning of what is stored in it
like this
```
code_worked = 9
Op.SSTORE(code_worked, 1)
...
```

Reply:
Done.  `ID_<calltype>` identifies what call type the result is tracking.